plugins/builds/index.js

```diff
@@ -870,19 +1058,77 @@ const buildsPlugin = {
                     });
 
                     if (existNextBuild === null) {
+                        const nextBuildNode = current.event.workflowGraph.nodes.find(
+                            node => node.name === trimJobName(nextJobName)
+                        );
+                        const nextBuildStageName = nextBuildNode.stageName || undefined;
+
+                        // Next job is in a stage
+                        if (nextBuildStageName) {
+                            const nextBuildStage = await stageFactory.get({
+                                pipelineId: current.pipeline.id,
+                                name: nextBuildStageName
+                            });
+                            const stageSetupName = getFullStageJobName({
+                                stageName: nextBuildStageName,
+                                jobName: 'setup'
+                            });
+
+                            // If next build is stage setup, create stageBuild and set to running
+                            if (nextBuildStage && nextJobName === stageSetupName) {
+                                await stageBuildFactory.create({
+                                    stageId: nextBuildStage.id,
+                                    eventId: current.event.id,
+                                    status: 'CREATED'
+                                });
+                            }
+                        }
+
```

Comment:
We should abstract this into `buildFactory`. This would ensure, `stageBuild` will always gets created when `setup` build is created irrespective of from where the API is called.

Reply:
Opened a PR: https://github.com/screwdriver-cd/models/pull/600